### PR TITLE
stm32: rework spi, based on PR #740 and #742.

### PR DESCRIPTION
**Pull request by grevaillot**
ok, so i did a quick pr based on #742 and #740 comments and previous work from @SevenW - this is more a draft than anything else yet, but seems a bit nicer than current master tree state. l0 spi is now enabled too.

This split spi stuff in three part:
- v1 : basic spi peripheral
- v1_frf : v1 spi with frf mode additional bit in spi_cr2 / spi_sr
- v2 : spi with variable datasize, fifo and other fancy stuff.

Then,
- v1 maps to f1 chips
- v1_frf to f2, f4 and l0,l1
- v2 to f0, f3 and l4

Note: This **breaks** spi_master_init API for v2 devices : function prototype from common spi header used to be abused, with DFF bit reused for CRCL bit.
Now, v2 spi_master_init does not handle anymore CRCL bits, as it does not usually mess with other crc configuration. I'm not exactly happy of that, as I keeping a consistent api between devices is nice, but DFF just does not exist on f0,f3 and l4,

### CODE REVIEW

File: lib/stm32/common/spi_common_v2.c [hunk 1]
```diff
@@ -59,40 +59,35 @@ LSB first.
...
 
 	reg32 |= br;		/* Set baud rate bits. */
 	reg32 |= cpol;		/* Set CPOL value. */
 	reg32 |= cpha;		/* Set CPHA value. */
-	reg32 |= crcl;		/* Set crc length (8 or 16 bits). */
 	reg32 |= lsbfirst;	/* Set frame format (LSB- or MSB-first). */
 
-	/* TODO: NSS pin handling. */
-
+	SPI_CR2(spi) |= SPI_CR2_SSOE;
```
Comment (karlp) [inline lib/stm32/common/spi_common_v2.c]: this is out of scope of this commit, and I'll probably split it locally.  It's consistent with what the v1 code does, ie, it's _right_ but it's not part of the refactor here :)

**Comment by karlp** [comment]
what do you think about dropping the dff altogether from spi init master v1? then they can both be same api

**Comment by grevaillot** [comment]
I'm all for it. I guess it could be possible to do something to but back spi_init_master into spi_common, i'll have a look.

**Comment by grevaillot** [comment]
So, this could work but, i'm wondering if (why?) we really need to keep the CRC bits untouched in spi_init_master while reseting the whole register. Also, do we clear DFF bit ? Current logic clears everything except CRC related stuff and SPIE.

Currently we would need to add a not-so-nice mask, I would prefer to avoid that, ie : https://github.com/grevaillot/libopencm3/commit/4e805dc48716681e8124a65ea232bc9a747b6fb3

**Comment by karlp** [comment]
I've merged this for now to get a groundwork going forward.

**Pull request closed without merging**